From fbbbccd00ea91b506e6e9430dc80734b8b2fdef6 Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <andrewgunnerson@gmail.com>
Date: Fri, 20 Dec 2019 02:54:57 -0500
Subject: [PATCH] ngx_rtmp_hls_cleanup_dir: Only delete files older than 5
 times the playlist length

The overhead of HLS meant that only keeping two times the playlist
length caused in-use .ts files and even the playlist itself to be
deleted.

Signed-off-by: Andrew Gunnerson <andrewgunnerson@gmail.com>
---
 hls/ngx_rtmp_hls_module.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hls/ngx_rtmp_hls_module.c b/hls/ngx_rtmp_hls_module.c
index fbf2bcb..398549a 100644
--- a/hls/ngx_rtmp_hls_module.c
+++ b/hls/ngx_rtmp_hls_module.c
@@ -2323,7 +2323,7 @@ ngx_rtmp_hls_cleanup_dir(ngx_str_t *ppath, ngx_msec_t playlen)
                              name.data[name.len - 2] == 't' &&
                              name.data[name.len - 1] == 's')
         {
-            max_age = playlen / 500;
+            max_age = playlen / 200;
 
         } else if (name.len >= 5 && name.data[name.len - 5] == '.' &&
                                     name.data[name.len - 4] == 'm' &&
@@ -2331,14 +2331,14 @@ ngx_rtmp_hls_cleanup_dir(ngx_str_t *ppath, ngx_msec_t playlen)
                                     name.data[name.len - 2] == 'u' &&
                                     name.data[name.len - 1] == '8')
         {
-            max_age = playlen / 500;
+            max_age = playlen / 200;
 
         } else if (name.len >= 4 && name.data[name.len - 4] == '.' &&
                                     name.data[name.len - 3] == 'k' &&
                                     name.data[name.len - 2] == 'e' &&
                                     name.data[name.len - 1] == 'y')
         {
-            max_age = playlen / 500;
+            max_age = playlen / 200;
 
         } else {
             ngx_log_debug1(NGX_LOG_DEBUG_RTMP, ngx_cycle->log, 0,
-- 
2.23.0

